src/components/withCurrentReportID.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {createContext, forwardRef, useCallback, useState, useMemo, RefAttributes, ComponentType, ForwardedRef} from 'react';
+import PropTypes from 'prop-types';
+import {NavigationState} from '@react-navigation/native';
+
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import Navigation from '../libs/Navigation/Navigation';
+
+type CurrentReportIDContextValue = {
+    updateCurrentReportID: (state: NavigationState) => void;
+    currentReportID: string;
+};
+type CurrentReportIDContextProviderProps = {
+    /** Actual content wrapped by this component */
+    children: React.ReactNode;
+};
+
+const CurrentReportIDContext = createContext<CurrentReportIDContextValue | null>(null);
+
+// TODO: Remove when depended components are migrated to TypeScript.
+const withCurrentReportIDPropTypes = {
+    /** Function to update the state */
+    updateCurrentReportID: PropTypes.func.isRequired,
+
+    /** The top most report id */
+    currentReportID: PropTypes.string,
+};
+
+const withCurrentReportIDDefaultProps = {
+    currentReportID: '',
+};
+
+function CurrentReportIDContextProvider(props: CurrentReportIDContextProviderProps) {
+    const [currentReportID, setCurrentReportID] = useState('');
+
+    /**
+     * This function is used to update the currentReportID
+     * @param state root navigation state
+     */
+    const updateCurrentReportID = useCallback(
+        (state: NavigationState) => {
+            setCurrentReportID(Navigation.getTopmostReportId(state) ?? '');
```

Comment:
why is this change needed?
I think it's original logic to return undefined or null as is

Reply:
@situchan So 
https://github.com/Expensify/App/blob/a094cbb4f02ffeb21608176d986de7b8ace1794d/src/components/withCurrentReportID.js#L20-L22
The default state is empty string so useState is infering from this that currentReportID will be always string, and 
`Navigation.getTopmostReportId` is returning `string | undefined` we need to use `??` to put empty string in case 
`Navigation.getTopmostReportId`  would return undefined